src/dnaio/_core.pyx

```diff
@@ -298,41 +299,39 @@ cdef bytes create_fastq_record(char * name, char * sequence, char * qualities,
         retval_ptr[cursor] = b"\n"
         return retval
 
-# It would be nice to be able to have the first parameter be an
-# unsigned char[:] (memory view), but this fails with a BufferError
-# when a bytes object is passed in.
-# See <https://stackoverflow.com/questions/28203670/>
 
-ctypedef fused bytes_or_bytearray:
-    bytes
-    bytearray
-
-
-def paired_fastq_heads(bytes_or_bytearray buf1, bytes_or_bytearray buf2, Py_ssize_t end1, Py_ssize_t end2):
+def paired_fastq_heads(buf1, buf2, Py_ssize_t end1, Py_ssize_t end2):
```

Comment:
If you see my answer on https://stackoverflow.com/questions/28203670/, it is actually possible now to declare the parameters to be of type `const unsigned char[:]`. It would be nicer to let Cython handle getting the memory view.